solidity/contracts/VerifierModule.sol

```diff
@@ -0,0 +1,279 @@
+      _arbitraryTxnParams.refundReceiver,
+      _arbitraryTxnParams.signatures
+    );
+
+    // Pay incentives
+    // TODO: Calculations for incentives so its not hardcoded to 1e18
+    ISafe(_safe).execTransactionFromModule(msg.sender, 1e18, '', Enum.Operation.Call);
+
+    // Make the storage updates at the end of the call to save gas in a revert scenario
+    latestVerifiedSettings[_safe] = _hashedProposedSettings;
+    latestVerifiedSettingsTimestamp[_safe] = block.timestamp;
+
+    emit VerifiedUpdate(_safe, _hashedProposedSettings);
+  }
+
+  /**
+   * @notice The function extracts the storage root of the StorageMirror contract from a given account proof
+   *
+   * @param _storageMirrorAccountProof The account proof of the StorageMirror contract from the latest block
+   */
+
+  function extractStorageMirrorStorageRoot(bytes memory _storageMirrorAccountProof)
+    external
+    view
+    returns (bytes32 _storageRoot)
+  {
+    (bytes memory _blockHeader,) = BLOCK_HEADER_ORACLE.getLatestBlockHeader();
+
+    StateVerifier.BlockHeader memory _parsedBlockHeader = StateVerifier.verifyBlockHeader(_blockHeader);
+
+    bytes memory _rlpAccount = MerklePatriciaProofVerifier.extractProofValue(
+      _parsedBlockHeader.stateRootHash,
+      abi.encodePacked(keccak256(abi.encode(STORAGE_MIRROR))),
+      _storageMirrorAccountProof.toRlpItem().toList()
+    );
+
+    _storageRoot = StateVerifier.extractStorageRootFromAccount(_rlpAccount);
+  }
+
+  /**
+   * @notice The function that verifies a given storage proof for the proposed settings
+   *
+   * @param _safe The address of the safe that has new settings
+   * @param _proposedSettings The new settings that are being proposed
+   * @param _storageMirrorStorageProof The storage proof of the StorageMirror contract on the home chain
+   * @return _hashedProposedSettings The keccak256 hash of the proposed settings
+   */
+
+  function _verifyNewSettings(
+    address _safe,
+    IStorageMirror.SafeSettings memory _proposedSettings,
+    bytes memory _storageMirrorStorageProof
+  ) internal view virtual returns (bytes32 _hashedProposedSettings) {
+    bytes32 _latestStorageRoot = STORAGE_MIRROR_ROOT_REGISTRY.latestVerifiedStorageRoot();
+
+    // The slot of where the latest settings hash is stored in the storage mirror
+    bytes32 _safeSettingsSlot = keccak256(abi.encode(_safe, _LATEST_VERIFIED_SETTINGS_SLOT));
+
+    bytes32 _safeSettingsSlotHash = keccak256(abi.encode(_safeSettingsSlot));
+
+    RLPReader.RLPItem[] memory _stack = _storageMirrorStorageProof.toRlpItem().toList();
+
+    bytes memory _slotValue =
+      MerklePatriciaProofVerifier.extractProofValue(_latestStorageRoot, abi.encodePacked(_safeSettingsSlotHash), _stack);
+
+    bytes32 _hashedSavedSettings = _bytesToBytes32(_slotValue);
+
+    _hashedProposedSettings = keccak256(abi.encode(_proposedSettings));
+
+    if (_hashedProposedSettings != _hashedSavedSettings) revert SettingsDontMatch();
```

Comment:
Same here, these are the complex parts of the project so it would be cool to have per-line small comments